lib/membrane_h264_ffmpeg/parser.ex

```diff
@@ -269,30 +270,19 @@ defmodule Membrane.H264.FFmpeg.Parser do
   end
 
   @impl true
-  def handle_stream_format(:input, %Membrane.H264.RemoteStream{}, ctx, _state)
+  def handle_stream_format(:input, %Membrane.H264.RemoteStream{} = stream_format, ctx, state)
       when ctx.pads.input.start_of_stream? do
-    raise "Cannot handle Membrane.H264.RemoteStream format after the stream has started"
+    Membrane.Logger.debug("Disposing old parser due to stream format change")
+
+    {[], state} = handle_init(%{}, state.original_opts)
+    {[], state} = handle_setup(%{}, state)
```

Comment:
Like above, this feels too hacky, especially mocking the context.
1. I'd create `Native.create!()` raising as in setup
2. Use that in `handle_setup` and here